src/combineReducers.js

```diff
@@ -134,6 +134,7 @@ export default function combineReducers(reducers) {
       nextState[key] = nextStateForKey
       hasChanged = hasChanged || nextStateForKey !== previousStateForKey
     }
+    hasChanged = hasChanged || Object.keys(state).length !== finalReducerKeys.length
```

Comment:
If the keys are the same length `hasChanged` should reflect the correct value. If we have an unknown key it'll be replaced with the missing reducer key.